File: test/timeConversions.py
import numpy as np
from datetime import datetime
from astropy import units as u
from astropy.time import Time
from astropy.coordinates import EarthLocation


# definitions
t1 = Time('2010-01-01 00:00:00', scale = 'utc')
t2 = Time('1987-08-23 10:25:00', scale = 'ut1')
location = EarthLocation(lat = 51.5 * u.deg, lon = 0.01 * u.deg)
tl1 = Time(t1, location = location)
tl2 = Time(t2, location = location)

# conversions
t1_jd = t1.jd
t2_jd = t2.jd
t1_mjd = t1.mjd
t2_mjd = t2.mjd
t1_ut1 = t1.ut1
t2_ut1 = t2.ut1
t1_utc = t1.utc
t2_utc = t2.utc
t1_lmst = tl1.sidereal_time('mean')
t2_lmst = tl2.sidereal_time('mean')
t1_last = tl1.sidereal_time('apparent')
t2_last = tl2.sidereal_time('apparent')
t1_gmst = tl1.sidereal_time('mean', 'greenwich')
t2_gmst = tl2.sidereal_time('mean', 'greenwich')
t1_gast = tl1.sidereal_time('apparent', 'greenwich')
t2_gast = tl2.sidereal_time('apparent', 'greenwich')



# printing conversions
print('==> t1 = ', t1)
print('  . t1 UTC = ', t1_utc)
print('  . t1 UT1 = ', t1_ut1)
print('  . t1 JD = ', t1_jd)
print('  . t1 MJD = ', t1_mjd)
print('  . t1 LMST = ', t1_lmst)
print('  . t1 LAST = ', t1_last)
print('  . t1 GMST = ', t1_gmst)
print('  . t1 GAST = ', t1_gast)

print('==> t2 = ', t2)
print('  . t2 UTC = ', t2_utc)
print('  . t2 UT1 = ', t2_ut1)
print('  . t2 JD = ', t2_jd)
print('  . t2 MJD = ', t2_mjd)
print('  . t2 LMST = ', t2_lmst)
print('  . t2 LAST = ', t2_last)
print('  . t2 GMST = ', t2_gmst)
print('  . t2 GAST = ', t2_gast)

File: test/timeConversions.jl
using Test

push!(LOAD_PATH, "$(@__DIR__)/../src")
using AstronomicalCoordinates



# ---------------------------------------------------------------------------------- #
#
# Comparisons below are with respect to AstroPy.
# ```
# 
# ```
@info "==> Time conversions"

t1 = TimeUTC(DateTime(2010, 01, 01, 00, 00, 00))
t2 = TimeUT1(DateTime(1987, 08, 23, 10, 25, 00))
t3 = TimeJD(2455197.5)
t4 = TimeMJD(47030.434)
t5 = TimeGMST(DateTime(1987, 08, 23, 10, 25, 00))
t6 = TimeGAST(DateTime(1987, 08, 23, 10, 25, 00))

# testName = "UTC, UT1 => JD, MJD"
# @info "$(testName)"
# @testset "$(testName)" begin
# 	t1_jd = convert(TimeJD, t1)
# 	t1_mjd = convert(TimeMJD, t1)
# 	t2_jd = convert(TimeJD, t2)
# 	t2_mjd = convert(TimeMJD, t2)
# 	@test t1_jd.value == 2455197.5
# 	@test t1_mjd.value == 55197.0
# 	@test t2_jd.value ≈ 2447030.93
# 	@test t2_mjd.value ≈ 47030.434
# end

# testName = " UT1 <=> UTC"
# @info "$(testName)"
# @testset "$(testName)" begin
# 	t1_ut1 = convert(TimeUT1, t1)
# 	t2_utc = convert(TimeUTC, t2)
# 	ut1 = DateTime(2010, 1, 1, 0, 0, 0, 114)
# 	utc = DateTime(1987, 08, 23, 10, 24, 59, 570) # astropy is 569 ms
# 	@test t1_ut1.value == ut1
# 	@test t2_utc.value == utc
# end

# testName = "JD <=> MJD"
# @info "$(testName)"
# @testset "$(testName)" begin
	t3_mjd = convert(TimeMJD, t3)
	t4_jd = convert(TimeJD, t4)
	@test t3_mjd.value == t3.value - AstronomicalCoordinates.mjd_0
	@test t4_jd.value == t4.value + AstronomicalCoordinates.mjd_0
# end

testName = "JD, MJD => GMST, GAST"
@info "$(testName)"
@testset "$(testName)" begin

	# t3_gast = convert(TimeGAST, t3)
	# t3_gmst = convert(TimeGMST, t3)
	# t4_gast = convert(TimeGAST, t4)
	t4_gmst = convert(TimeGMST, t4)
	
	# gast3 = DateTime(2010, 01, 01, 06, 42, 10, 150)
	# gast4 = DateTime(1987, 08, 23, 08, 29, 43, 587)
	gmst3 = DateTime(2010, 01, 01, 06, 42, 09, 144)
	gmst4 = DateTime(1987, 08, 23, 08, 29, 43, 584)

	# @show(t3_gmst.value)
	# @show(gmst3)
	@show(t4_gmst.value)
	@show(gmst4)
	# @show(t4_gast.value)
	# @show(gast4)

	# @test t3_gmst.value == gmst3
	# @test t3_gast.value == gast3
	# @test t4_gmst.value == gmst4
	# @test t4_gast.value == gast4
	
end

# testName = "GMST/GAST <=> LMST/LAST"
# @info "$(testName)"
# @testset "$(testName)" begin
# 	location = GeoCoordinatesGeocentric(50. * u"°", 10. * u"°")

# 	t5_lmst = convert(TimeLMST, t5, location)
# 	t6_last = convert(TimeLAST, t6, location)

# 	lmst5 = DateTime(1987, 08, 23, 8, 29, 46, 416)
# 	last6 = DateTime(1987, 08, 23, 8, 29, 46, 419)

# 	@test t5_lmst.value == lmst5
# 	@test t6_last.value == last6
	
# end



# ---------------------------------------------------------------------------------- #